### ISSUE TITLE
`openfl.net.FileReference` freezes the application after invoking `save`

### ISSUE THREAD

**Issue by DetectiveBaldi** [openfl/openfl#2742]
**Describe the bug**
`openfl.net.FileReference` causes the application to freeze after calling an upload and / or browsing method like `browse` or `save`

**To Reproduce**
The bug can be reproduced with the following code sample:
```haxe
package;

import openfl.display.Sprite;

import openfl.net.FileReference;

class Main extends Sprite
{
	public function new()
	{
		super();

		var fileRef:FileReference = new FileReference();

		fileRef.save(0, "path.txt");
	}
}
```
The application will freeze up either immediately after the browsing window is opened, or after a couple of seconds

**Expected behavior**
The application does not freeze when using a `FileReference`

**OpenFL Targets**
This bug was tested and reproduced on HL.

**Additional context**
This bug exists only on the latest git versions of Lime and OpenFL. Using stable Lime 8.1.3 and OpenFL 9.3.4 results in no application freeze.

**Comment by joshtynjala** [comment]
I'm having difficulty reproducing this issue as described.

Here's the code that I used (which includes some basic animation as a way to verify whether the app has frozen or not).

```haxe
import openfl.display.Sprite;
import openfl.events.Event;
import openfl.events.MouseEvent;
import openfl.net.FileReference;

class Main extends Sprite {
	public function new() {
		super();

		var ball = new Sprite();
		ball.graphics.beginFill(0xff0000);
		ball.graphics.drawCircle(20, 20, 20);
		ball.graphics.endFill();
		addChild(ball);
		var xOffset = 1.0;
		var yOffset = 1.0;
		addEventListener(Event.ENTER_FRAME, event -> {
			ball.x += xOffset;
			ball.y += yOffset;
			if (ball.x < 0.0 || ball.x + ball.width > stage.stageWidth) {
				xOffset = -xOffset;
			}
			if (ball.y < 0.0 || ball.y + ball.height > stage.stageHeight) {
				yOffset = -yOffset;
			}
		});

		stage.addEventListener(MouseEvent.CLICK, event -> {
			var fileRef:FileReference = new FileReference();
			fileRef.save(0, "path.txt");
		});
	}
}
```

| Versions | Target | OS | Result |
| -- | -- | -- | -- |
| 8.1.3/9.3.4 | Neko | macOS 14 | No freeze |
| 8.1.3/9.3.4 | HL | macOS 14 | **Freeze** |
| 8.1.3/9.3.4 | C++ | macOS 14 | No freeze |
| git/git | Neko | macOS 14 | No freeze |
| git/git | HL | macOS 14 | **Freeze** |
| git/git | C++ | macOS 14 | No freeze |
| 8.1.3/9.3.4 | Neko | Windows 11 | No freeze |
| 8.1.3/9.3.4 | HL | Windows 11 | **Freeze** |
| 8.1.3/9.3.4 | C++ | Windows 11 | No freeze |
| git/git | Neko | Windows 11 | No freeze |
| git/git | HL | Windows 11 | **Freeze** |
| git/git | C++ | Windows 11 | No freeze |

In my tests, I _always_ see a freeze when targeting HashLink, regardless of Lime/OpenFL versions and operating system used. However, other native targets (Neko and C++/Windows/macOS) never freeze.

This is definitely something that should be fixed with HashLink in the future, but it is apparently not a new issue based on these results. In the past, I seem to recall seeing some conditional compilation in Lime like `#if (cpp || neko)`, and if that still exists, it should probably be updated to `#if (cpp || neko || hl)` instead.

**Comment by joshtynjala** [comment]
Oh wait. I apologize. I can see that on Windows with Lime and OpenFL from git when targeting HashLink only (Neko and C++ on Windows are fine), even the file dialog stops responding to user input. I misunderstood what you meant by freeze because I had observed the animation freeze with HashLink. I see now that it's a different sort of freeze.

Yes, I can now reproduce this bug. But only when targeting HashLink on Windows.

**Comment by DetectiveBaldi** [comment]
I've had the opportunity to do some more testing, and I've found that in a fresh OpenFL project, the program only crashes while targeting HashLink, and I've updated the issue accordingly
I believe while making the test sample I skipped testing Windows, due to the fact that the same code crashed for my full project on all targets

**Comment by DetectiveBaldi** [comment]
I'm not sure why exactly it's crashing on my full project, it could have something to do with the fact I'm using Flixel, although I'm not quite sure why that would interfere.

**Comment by DetectiveBaldi** [comment]
My main question is why are the stable releases of Lime and OpenFL not freezing as expected? Even on HL, I experience no application freeze with Lime 8.1.3 and OpenFL 9.3.4

**Comment by joshtynjala** [comment]
> why are the stable releases of Lime and OpenFL not freezing as expected?

It's the switch from BackgroundWorker to ThreadPool in Lime's FileDialog. More investigation is needed on the Lime side to figure out why. I opened an issue over there: openfl/lime#1849

**Comment by DetectiveBaldi** [comment]
with the new stable releases of lime and openfl, it seems my issue has been resolved across all targets. perhaps i made a mistake when rebuilding the libraries myself from git

**Comment by joshtynjala** [comment]
> perhaps i made a mistake when rebuilding the libraries myself from git

You did not make a mistake. We included a workaround for this issue in Lime 8.2.0.
